File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s) {
    string answer = "";
    int j=0;
    for(int i=0; i<s.size(); i++){
        
        if(s[i] == ' '){
            j=0;
            continue;
        }
        if(j%2==0){
            s[i] = toupper(s[i]);
            j++;
        } else {
            s[i] = tolower(s[i]);
            j++;
        }
        
        
    }
    
    
    return s;
}

File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.java
class Solution {
    public String solution(String s) {
        
        StringBuilder sb = new StringBuilder();
        String[] arr = s.split(" ",-1);
        
        for(String word: arr){
            
            System.out.println(word);
            if(word.equals(" ")){
                sb.append(" ");
                continue;
            }
            
            StringBuilder str = new StringBuilder();
            
            for(int j=0; j<word.length(); j++){
                
                char ch = word.charAt(j);
                if(j%2==0) {
                    str.append(Character.toUpperCase(ch));
                } else {
                    str.append(Character.toLowerCase(ch));
                }
            }
            sb.append(str).append(" ");
        }
        
        return sb.toString().substring(0,sb.length()-1);
        
        
        
        
//         String answer = "";
//         int index = 0;
        
//         for(int i=0; i<s.length(); i++){
            
//             char ch = s.charAt(i);
            
//             if(ch == ' '){
//                 answer += " ";
//                 index= 0;
//                 continue;
//             }
            
//             if(index %2==0){
//                 answer += Character.toUpperCase(ch);
//             } else{
//                 answer += Character.toLowerCase(ch);
//             }
//             index++;
            
//         }
        
        
//         return answer;
    }
}